operator/v1/types_console.go

```diff
@@ -375,6 +380,34 @@ const (
 	BrandROSA Brand = "ROSA"
 )
 
+// Ingress allows cluster admin to configure alternative ingress for the console.
+type Ingress struct {
+	// consoleURL is a URL to be used as the base console address.
+	// If not specified, the console route hostname will be used.
+	// This field is required for clusters without ingress capability,
+	// where access to routes is not possible.
+	// Make sure that appropriate ingress is set up at this URL before assigning the value,
+	// otherwise the console operator may go degraded.
```

Comment:
Feels like there's tension between being required for clusters without `Ingress` (with no timing notes) and needing a functional target.  I understand that the console operator wants a functional `consoleURL`.  And that when `Ingress` is enabled, it can provide that default console URL itself.  But if you're in a cluster without `Ingress`, you probably want to set `consoleURL` to point at where you expect the console to be (as soon as you know where that is), without worrying about delaying until you know it's working, right?  That way the console operator can check whether it's working for you, and let you know if you have more work to do.  Can we reword this (and the similar `clientDownloadsURL` wording) to something like:

> .... This field is required for clusters without Ingress capability, where access to Routes is not possible.  The console operator will monitor the target and may set its ClusterOperator Degraded=True if the target is unreachable for sufficiently long to represent a lower quality of service.

which lifts some "lower quality of service" wording from [the `Degraded` Godocs][1].  Alternatively, just leave it implicit that the console operator is on top of this, and stop after "where access to routes is not possible."?

[1]: https://github.com/openshift/api/blob/0314f31e77748a7bdfa7c02286193760dee85195/config/v1/types_cluster_operator.go#L175-L178

Reply:
I see some discussion of this wording around [here][1] (sorry, I hadn't gone through `Outdated` but still unresolved threads before posting 😅), so feel free to ignore if I'm just missing context that's already been hashed out.

[1]: https://github.com/openshift/api/pull/1887#discussion_r1608003797

Reply:
> But if you're in a cluster without Ingress, you probably want to set consoleURL to point at where you expect the console to be (as soon as you know where that is), without worrying about delaying until you know it's working, right? That way the console operator can check whether it's working for you, and let you know if you have more work to do

Good point. Indeed, the admin cannot be sure the URL is working until it gives the custom URL to the console operator. Since the operator has to update the internal and public configs before the full flow can work. Let me think of how I can reflect this in the description.

Reply:
Rephrased to:
```
    // consoleURL is a URL to be used as the base console address.
    // If not specified, the console route hostname will be used.
    // This field is required for clusters without ingress capability,
    // where access to routes is not possible.
    // Make sure that appropriate ingress is set up at this URL.
    // The console operator will monitor the URL and may go degraded
    // if it's unreachable for an extended period.                                                                                    
    // Must use the HTTPS scheme.
```

Reply:
I don't have the ability to mark this thread resolved myself, but the c0622aa4e679d Godocs look good to me :bow: